src/main/java/com/kadai10/employee/service/EmployeeService.java

```diff
@@ -45,7 +46,9 @@ public List<Employee> findByName(String name) {
      * @param id 取得したいユーザーのid
      * @return 指定されたIDに対応するユーザー情報。存在しない場合は空のOptionalを返す
      */
-    public Optional<Employee> findById(Integer id){ return employeeMapper.findById(id);}
+    public Optional<Employee> findById(Integer id){
+        return Optional.ofNullable(employeeMapper.findById(id).orElseThrow(() -> new UserNotFoundException("userId " + id +" could not be found")));
+    }
```

Comment:
ユーザーが見つからなかった場合、UserNotFoundExceptionをスローする処理になったので、ここは無理にOptionalを返却する必要はなくなってます！
こんな感じで良いと思いますよ！
この修正を入れると、EmployeeService#findByIdを呼び出してるContoroller側も修正が必要になると思いますので追従して修正してあげましょう！
```suggestion
    public Employee findById(Integer id){
        return employeeMapper.findById(id).orElseThrow(() -> new UserNotFoundException("userId " + id +" could not be found"));
    }
```

Reply:
お騒がせしてすみません。
6190c1081be22e32674364841ed2e39eed719ea6
Contoroller側も同じように修正したら、Optionalを返却せずに実装できるようになりました。